samples/dnn/classification.cpp

```diff
@@ -102,39 +94,66 @@ int main(int argc, char** argv)
     CV_Assert(!model.empty());
 
     //! [Read and initialize network]
-    Net net = readNet(model, config, framework);
+    Net net = readNetFromONNX(model);
     net.setPreferableBackend(backendId);
     net.setPreferableTarget(targetId);
-    //! [Read and initialize network]
 
     // Create a window
     static const std::string kWinName = "Deep learning image classification in OpenCV";
-    namedWindow(kWinName, WINDOW_NORMAL);
+    cv::namedWindow(kWinName, WINDOW_NORMAL);
+
 
-    //! [Open a video file or an image file or a camera stream]
     VideoCapture cap;
-    if (parser.has("input"))
-        cap.open(parser.get<String>("input"));
-    else
-        cap.open(0);
-    //! [Open a video file or an image file or a camera stream]
+    vector<string> imageFiles;
+    size_t currentImageIndex = 0;
+
+    if (parser.has("input")) {
+        string input = parser.get<String>("input");
+
+        if (input.find('.')==string::npos) {
+            // Input is a directory, list all image files
+            imageFiles = listImageFilesInDirectory(input);
+            if (imageFiles.empty()) {
+                cout << "No images found in the directory." << endl;
+                return -1;
+            }
+        } else {
+            // Input is not a directory, try to open as video or image
+            cap.open(input);
+            if (!cap.isOpened()) {
+                cout << "Failed to open the input." << endl;
+                return -1;
+            }
+        }
+    } else {
+        cap.open(0); // Open default camera
+    }
+
 
     // Process frames.
     Mat frame, blob;
-    while (waitKey(1) < 0)
+    while (true)
```

Comment:
some compilers complain about `while (true) {...}`. Use `for (;;) {...}` instead